binary_search_tree/binary_search_tree.py

```diff
@@ -5,13 +5,53 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    pass
+    while self:
+      if value < self.value and self.left is None:
+        self.left = BinarySearchTree(value)
+        break
+      elif value < self.value and self.left is not None:
+        self = self.left
+      elif value >= self.value and self.right is None:
+        self.right = BinarySearchTree(value)
+        break
+      else:
+        self = self.right
+    # The recursive version below also works.
+    # if value < self.value and self.left is None:
+    #   self.left = BinarySearchTree(value)
+    # elif value < self.value and self.left is not None:
+    #   self.left.insert(value)
+    # elif value >= self.value and self.right is None:
+    #   self.right = BinarySearchTree(value)
+    # else:
+    #   self.right.insert(value)
 
   def contains(self, target):
-    pass
+    while True:
+      if self.value == target:
+        return True
+      elif target < self.value and self.left is None:
+        return False
+      elif target < self.value and self.left is not None:
+        self = self.left
+      elif target >= self.value and self.right is None:
+        return False
+      else:
+        self = self.right
 
   def get_max(self):
-    pass
+    while self:
+      if self.right is None:
+        return self.value
+      else:
+        self = self.right
 
   def for_each(self, cb):
-    pass
\ No newline at end of file
+    cb(self.value)
+    if self.left and self.right:
+      self.left.for_each(cb)
+      self.right.for_each(cb)
+    elif self.left:
+      self.left.for_each(cb)
+    elif self.right:
+      self.right.for_each(cb)
```

Comment:
I'm curious if your tests would still pass if you only had two if statements. if self.left and if self.right.